pkg/secrets/secrets.go

```diff
@@ -0,0 +1,30 @@
+package secrets
+
+import "os"
+
+type Getter interface {
+	Get(key string) (string, error)
+}
+
+type Backend struct {
+	getter Getter
+}
+
+// NewSecretBackend creates a new SecretBackend based on a concrete secrets.Getter implementation.
+func NewSecretBackend(getter Getter) *Backend {
+	return &Backend{getter: getter}
+}
+
+// Get fetches a secret from the implemented SecretBackend.
+func (b Backend) Get(key string) (string, error) {
+	return b.getter.Get(key)
+}
```

Comment:
I think we should make this private for now since it's not used by any implementations.